src/System.Threading.Timer/tests/TimerFiringTests.cs

```diff
@@ -93,6 +93,7 @@ public void MultipleTimers_ShouldFire_InCorrectOrder()
             if (count == 0)
             {
                 Assert.Same(t2State, o);
+                Interlocked.Increment(ref count);
```

Comment:
And thinking about it more, you might just want to delete this whole test.  Technically there's nothing guaranteeing that t2 will fire before t1.